pulsar/impl_consumer.go

```diff
@@ -154,27 +155,47 @@ func (c *consumer) Unsubscribe() error {
 }
 
 func (c *consumer) Receive(ctx context.Context) (Message, error) {
+	if len(c.consumers) > 1 {
+		for _, pc := range c.consumers {
+			go func(pc Consumer) {
+				for {
+					msg, err := pc.Receive(ctx)
+					if err != nil {
+						return
+					}
+
+					consumerMessage := ConsumerMessage{
+						Message:  msg,
+						Consumer: pc,
+					}
+					c.queue <- consumerMessage
+				}
+			}(pc)
+		}
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		case cMsg, ok := <-c.queue:
+			if ok {
+				return cMsg.Message, nil
+			}
+			return nil, errors.New("receive message error")
+		}
+	}
+
+	return c.consumers[0].(*partitionConsumer).Receive(ctx)
+}
+
+func (c *consumer) ReceiveAsync(ctx context.Context, msgs chan<- ConsumerMessage) error {
```

Comment:
Please add a receiveAsync method, that returns only one message, and following the implementation of Java client.